scripts/src/public/services/orders.ts

```diff
@@ -144,12 +145,13 @@ export async function createMarketplaceOrder(offerId: string, user: User, logger
 }
 
 async function createP2PExternalOrder(sender: User, jwt: ExternalPayToUserOrderJwt): Promise<db.ExternalOrder> {
-	const recipient = await User.findOneById(jwt.recipient.user_id);
+	const recipient = await User.findOne({ appId: sender.appId, appUserId: jwt.recipient.user_id });
 
 	if (!recipient) {
-		throw new Error(`couldn't find user with id "${ jwt.recipient.user_id }"`);
+		throw NoSuchUser(jwt.recipient.user_id);
 	}
 
+	await addWatcherEndpoint([recipient.walletAddress]);
```

Comment:
Great, we should add what Ron started to work on with dynamic watchers on payment service